core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -359,6 +401,41 @@ class TiBatchWrite(@transient val df: DataFrame,
     }
   }
 
+  private def lockTable(): Unit = {
+    if (isEnableTableLock) {
+      if (!tableLocked) {
+        tiDBJDBCClient.lockTableWriteLocal(options.database, options.table)
+        tableLocked = true
+      } else {
+        logger.warn("table already locked!")
+      }
+    } else {
+      // TODO: what if version of tidb does not support lock table
```

Comment:
In this case, just throw an exception and abort this operation.

Reply:
i will do it in this issue

https://github.com/pingcap/tispark/issues/839